odoo/addons/base/static/src/js/res_config_settings.js

```diff
@@ -293,7 +293,7 @@ var BaseSettingController = FormController.extend({
      */
     _onButtonClicked: function (ev) {
         var self = this;
-        if (ev.data.attrs.name !== 'execute' && ev.data.attrs.name !== 'cancel') {
+        if (ev.data.attrs.name !== 'execute' && ev.data.attrs.name !== 'cancel' && ev.data.attrs.discard !== '0') {
```

Comment:
Doesn't seem to be the right fix. If the user has made any other changes, you'll just overwrite them without warning.